### ISSUE TITLE
RelPositionalEncoding implementation has some bug？

### ISSUE THREAD

**Issue by cindycindyhi** [wenet-e2e/wenet#1611]
**Describe the bug**
The relposencoding implementation is same with abs pos encoding. 

**Implemention now**
<pre>
  self.pe[:, 0::2] = torch.sin(position * div_term)
  self.pe[:, 1::2] = torch.cos(position * div_term)
 </pre>

**Expected behavior**
The relposencoding in conformer descriped in paper should be: 
<pre>
  pe_positive[:, 0::2] = torch.sin(position * div_term)  
  pe_positive[:, 1::2] = torch.cos(position * div_term)  
  pe_negative[:, 0::2] = torch.sin(-1 * position * div_term)  
  pe_negative[:, 1::2] = torch.cos(-1 * position * div_term)  
  pe = torch.cat([pe_positive, pe_negative], dim=1)  
 </pre>

**Comment by robin1001** [comment]
Yes, it's not exactly the same to the paper, we just use it.

**Comment by cindycindyhi** [comment]
I've experimented and got better wer with the standard relative positional encoding. 1% improvement.

**Comment by Mddct** [comment]
> I've experimented and got better wer with the standard relative positional encoding. 1% improvement.

dynamic chunk training or non streamming？

**Comment by cindycindyhi** [comment]
> > I've experimented and got better wer with the standard relative positional encoding. 1% improvement.
> 
> dynamic chunk training or non streamming？
> 
> in origin transformer paper pos emb can be like pe = torch.cat([pe_positive, pe_negative], dim=1)

Non streaming with business data in real scenarios, not open source data.

**Comment by xingchensong** [comment]
Will hold this PR for anyone else who has encountered and interested in the same question.
